pkg/api/bytes.go

```diff
@@ -71,9 +72,15 @@ func (s *server) bytesGetHandler(w http.ResponseWriter, r *http.Request) {
 		jsonhttp.BadRequest(w, "invalid root chunk")
 		return
 	}
+	type targetsContextKey string
+	k := targetsContextKey("targets")
+
+	targets := r.URL.Query().Get("targets")
+	r = r.WithContext(context.WithValue(r.Context(), k, targets))
```

Comment:
it is good that you're not using the `string` type for the `WithValue` call, but a new type should not be defined inside of this function.

you can try something like:

```golang
type contextKey string

var (
    targetsContextKey = contextKey("targets")
)
```

alternatively, you can take a look at the `retrieval` and `tracing` packages which define context keys through the use of `struct` and maybe use this as inspiration.

Reply:
Good point, 
I extracted this out of the function.
It can be used as a struct, no need to define as a string.